src/core/action-creator.ts

```diff
@@ -196,8 +196,8 @@ export default class Creator {
         dispatch(this.receiveTemplate(Adapters.Search.extractTemplate(results.template))),
       );
 
-      Promise.all(updates)
-        .then(() => this.saveState());
+      return Promise.all(updates)
+        .then(() => this.saveState('search'));
```

Comment:
I'd suggest we pull out `'search'` and the other routes everywhere they're used into a constant somewhere
(this includes places where it's used at the key in a map, like in the `mergeState` map)